src/main/java/core/basesyntax/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int EXPECTED_FIELDS_LENGTH = 3;
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final String COMMA = ",";
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        if (data == null) {
+            throw new NullPointerException("Input data cannot be null");
```

Comment:
IllegalArgumentException is more appropriate exception in this case
```suggestion
            throw new IllegalArgumentException("Input data cannot be null");
```